app/src/main/java/org/mozilla/focus/tabs/TabModelStore.java

```diff
@@ -40,27 +42,34 @@ public static TabModelStore getInstance(@NonNull final Context context) {
         return instance;
     }
 
-    public void getSavedTabs(AsyncQueryListener listener) {
-        new QueryTabsTask(tabsDatabase, listener).executeOnExecutor(SERIAL_EXECUTOR);
+    public void getSavedTabs(@NonNull final Context context, final AsyncQueryListener listener) {
+        new QueryTabsTask(context, tabsDatabase, listener).executeOnExecutor(SERIAL_EXECUTOR);
     }
 
-    public void saveTabs(final List<TabModel> tabModelList, AsyncSaveListener listener) {
-        if (tabModelList == null || tabModelList.size() == 0) {
+    public void saveTabs(@NonNull final Context context, @NonNull final List<TabModel> tabModelList, @NonNull final String currentTabId, final AsyncSaveListener listener) {
+        if (tabModelList.size() == 0) {
             if (listener != null) {
                 listener.onSaveComplete();
             }
             return;
         }
 
+        PreferenceManager.getDefaultSharedPreferences(context)
+                .edit()
+                .putString(context.getResources().getString(R.string.pref_key_current_tab_id), currentTabId)
+                .apply();
+
```

Comment:
I am not sure that whether Preference is a good place to persist small piece of data, or to persist in DB. It is up to you.